src/profile-logic/merge-compare.js

```diff
@@ -217,7 +217,22 @@ export function mergeProfilesForDiffing(
 
     // We adjust the various times so that the 2 profiles are aligned at the
     // start and the data is consistent.
-    const startTimeAdjustment = -thread.samples.time[0];
+    let startTimeAdjustment = 0;
+    if (thread.samples.length !== 0) {
+      startTimeAdjustment = -thread.samples.time[0];
+    }
+    if (thread.markers.length !== 0) {
```

Comment:
For consistency with https://github.com/firefox-devtools/profiler/blob/756250859908d5fd480b09c0498b7f2cc97b178c/src/profile-logic/profile-data.js#L1084 I'd use a `else if` instead of using both always. What do you think?

(this should also fix the failing test)